add release command

TheoBrigitte opened the pull request:
Towards: https://github.com/giantswarm/giantswarm/issues/5206

Provide `architect release` command, whenever a git tag is present, it creates a draft github release, with helm charts uploaded as assets.

xh3b4sd commented on cmd/release.go:
```diff
@@ -0,0 +1,87 @@
+}
+
+func runRelease(cmd *cobra.Command, args []string) {
+	var err error
```
Can we please have something like calling `runReleaseError` here and have proper error handling like anywhere else plus proper logging configured? From a software engineering point of view it would be really beneficial holding up our high standards and not writing random code. We do not have to reinvent the wheel and maintaining such different code will only be more complicated down the road. You may also want to read https://github.com/giantswarm/giantswarm/blob/master/sigs/operator/command_line_tools.md. We could also think about extending the docs with more pitfalls we find here.

kopiczko replied:
I'd also like to have program termination handled in a single place (or as few as possible).

kopiczko commented on cmd/release.go:
```diff
@@ -0,0 +1,87 @@
+		if dryRun {
+			log.Printf("dry run, not actually running workflow")
+			return
+		}
```
This dry-run feels useless, but I guess it's useless anyway.

TheoBrigitte commented:
Restructured the release command:
* moved into `cmd/release` package
* using cobra RunE in order to exit program in a single place
* added errors definition

TheoBrigitte commented on cmd/root.go:
```diff
@@ -80,32 +83,44 @@ func init() {
+	RootCmd.PersistentFlags().StringVar(&sha, "sha", defaultSha, "git SHA1 being built")
+	RootCmd.PersistentFlags().String("tag", defaultTag, "git tag being built")
+
+	RootCmd.PersistentFlags().String("version", defaultVersion, "project version being built")
```
Setting all values as flag is required in order to be able to access them from subcommands using `Command.Flags()`.

TheoBrigitte commented:
Need a second review round, please
